### PR TITLE
Plug external data checker

### PR DESCRIPTION
**Pull request by andrunko**
This uses the anitya checker to monitor updates:
- https://release-monitoring.org/project/13344/

Fixes: https://github.com/flathub/org.chromium.Chromium/issues/29

### CODE REVIEW

**Comment by andrunko** [comment]
Note that this will very likely produce PRs with conflicts when updating major versions, but should work fine in most cases at least for minor updates.

Some of the changes could arguably be done in flatpak-external-data-checker itself (e.g. to avoid rewriting coding style), but they are simple enough that I don't think it is worth it.

File: org.chromium.Chromium.metainfo.xml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-<!-- Copyright 2017 The Chromium Authors -->
```
Comment (wjt) [inline org.chromium.Chromium.metainfo.xml:1]: You can put it after the `<?xml ...?>` tag. I went to [great lengths](https://github.com/flathub/flatpak-external-data-checker/blob/8b5dfa2891b10502cc63804248634e1213a41396/src/lib/appdata.py#L92-L125) to make this work. [Here's the test](https://github.com/flathub/flatpak-external-data-checker/blob/8b5dfa2891b10502cc63804248634e1213a41396/tests/test_appdata.py#L192-L223).

**Review by wjt — CHANGES_REQUESTED** [review]
Apart from the build failure which looks to be related to moving a script out to an actual file, and one nitpick, this LGTM

**Comment by wjt** [comment]
> Some of the changes could arguably be done in flatpak-external-data-checker itself (e.g. to avoid rewriting coding style), but they are simple enough that I don't think it is worth it.

I went to some lengths to preserve YAML formatting, [e.g.](https://github.com/flathub/flatpak-external-data-checker/blob/8b5dfa2891b10502cc63804248634e1213a41396/src/lib/utils.py#L280-L284), but reached a point where I couldn't go much further without disproportionate effort :-)

File: org.chromium.Chromium.metainfo.xml (hunk 1 above)
Comment (andrunko) [reply org.chromium.Chromium.metainfo.xml:1]: Thanks, updated - although not sure how accurate this copyright notice is.

**Comment by andrunko** [comment]
Pushed some changes to try fixing the build, let's see (I was testing the scripts manually but forgot to test integration with the manifest...).

File: org.chromium.Chromium.metainfo.xml (hunk 1 above)
Comment (wjt) [reply org.chromium.Chromium.metainfo.xml:1]: Agreed. But at least it's still here so it could be updated :-)

**Comment by wjt** [comment]
I tested this locally, and I got:

```
+ 2021-01-06 22:31:30,090 WARNING src.checkers.htmlchecker: https://commondatastorage.googleapis.com/chromium-browser-official/chromium-87.0.4280.141.tar.xz returned HTTP Error 404: Not Found
BROKEN: chromium-87.0.4280.88.tar.xz
 Unreachable URL: https://commondatastorage.googleapis.com/chromium-browser-official/chromium-87.0.4280.88.tar.xz
```

It seems the new tarball hasn't appeared yet, but also that something weird is going on with the old URL. Maybe it's just a bug in the checker. Presumably once the new tarball appears the checker would succeed.

**Comment by andrunko** [comment]
> I tested this locally, and I got:
> 
> ```
> + 2021-01-06 22:31:30,090 WARNING src.checkers.htmlchecker: https://commondatastorage.googleapis.com/chromium-browser-official/chromium-87.0.4280.141.tar.xz returned HTTP Error 404: Not Found
> BROKEN: chromium-87.0.4280.88.tar.xz
>  Unreachable URL: https://commondatastorage.googleapis.com/chromium-browser-official/chromium-87.0.4280.88.tar.xz
> ```
> 
> It seems the new tarball hasn't appeared yet, but also that something weird is going on with the old URL. Maybe it's just a bug in the checker. Presumably once the new tarball appears the checker would succeed.

Yes, the new tarball hasn't appeared yet indeed (funny that I tested earlier and there was no `.141` announced yet). 
The error seems to be an issue with the `Alitya` checker that inherits from `HTMLChecker` and uses `HTMLChecker._update_version()` to validate the new URL which then sets it to `BROKEN` as it can't fetch it (404). The debug output is wrong because the summary uses the URL for the current version for `BROKEN` data, but it should be a non-issue.

I guess we have to live with the fact that release-monitoring may get updated before the new release tarball is uploaded.

**Comment by refi64** [comment]
New version is up so I guess it's the perfect time to take this for a spin, any issues in the actual checker of course aren't critical to this itself

**Pull request merged**